src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckPolicyMessageContentCollector.java

```diff
@@ -62,6 +70,63 @@ public BlackDuckPolicyMessageContentCollector(final JsonExtractor jsonExtractor,
 
     @Override
     protected void addCategoryItems(final List<CategoryItem> categoryItems, final JsonFieldAccessor jsonFieldAccessor, final List<JsonField<?>> notificationFields, final NotificationContent notificationContent) {
+        if (NotificationType.POLICY_OVERRIDE.name().equals(notificationContent.getNotificationType())) {
+            processPolicyOverride(categoryItems, jsonFieldAccessor, notificationFields, notificationContent);
+        } else {
+            processPolicy(categoryItems, jsonFieldAccessor, notificationFields, notificationContent);
+        }
+    }
+
+    private void processPolicy(final List<CategoryItem> categoryItems, final JsonFieldAccessor jsonFieldAccessor, final List<JsonField<?>> notificationFields, final NotificationContent notificationContent) {
+        final List<JsonField<PolicyInfo>> policyFields = getFieldsOfType(notificationFields, new TypeRef<PolicyInfo>() {});
+        final List<JsonField<ComponentVersionStatus>> componentFields = getFieldsOfType(notificationFields, new TypeRef<ComponentVersionStatus>() {});
+        try {
+            final List<PolicyInfo> policyItems = getFieldValueObjectsByLabel(jsonFieldAccessor, policyFields, BlackDuckProviderContentTypes.LABEL_POLICY_INFO_LIST);
+            final List<ComponentVersionStatus> componentVersionStatuses = getFieldValueObjectsByLabel(jsonFieldAccessor, componentFields, BlackDuckProviderContentTypes.LABEL_COMPONENT_VERSION_STATUS);
+            final Map<String, SortedSet<LinkableItem>> policyItemMap = mapPolicyToComponents(componentVersionStatuses);
+
+            final ItemOperation operation = getOperationFromNotification(notificationContent);
+            if (operation == null) {
```

Comment:
We should do this before the json parsing to save time if we do end up "short-circuiting".